source/app/dumping.cpp

```diff
@@ -672,6 +743,103 @@ void dumpOnlineFiles() {
     else showDialogPrompt(L"Failed to dump the online files...", L"OK");
 }
 
+void dumpSpotpass() {
+    std::vector<std::shared_ptr<titleEntry>> queue;
+    dumpingConfig spotpassConfig = { .dumpTypes = (DUMP_TYPE_FLAGS::CUSTOM) };
+
+    if (!showOptionMenu(spotpassConfig, false)) return;
+
+    titleEntry slcSpotpassDir{ .shortTitle = L"SpotPass Directory", .customFolder = folderPart{.inputPath = convertToPosixPath("/vol/storage_mlc01/usr/save/system/boss"), .outputPath = "/SpotPass Files"} };
+    queue.emplace_back(std::make_shared<titleEntry>(slcSpotpassDir));
+
+    dumpFileFilter filter{ .fileNames = {"task.db"} };
+    if (dumpQueue(queue, spotpassConfig, &filter)) showDialogPrompt(L"Successfully dumped all of the SpotPass files!", L"OK");
+    else showDialogPrompt(L"Failed to dump the SpotPass files...", L"OK");
+
+    wchar_t promptMessage[512];
+    swprintf(promptMessage, 512, L"Do you want to upload the files to the SpotPass Archival Project?\n" 
+                                  "The files do not contain any personally identifiable information.\n"
+                                  "It will upload %d files of 1MB each!\n", filter.outMatchedFiles.size());
+
+    uint8_t doUploadFiles = showDialogPrompt(promptMessage, L"Yes", L"No");
+    if (doUploadFiles == 0) {
+
+        size_t currentUploadIdx = 0;
+        bool taskUploadError = false;
+
+        struct userdata_upload_t {
+            bool* taskUploadError;
+            size_t* currentUploadIdx;
+            dumpFileFilter* filter;
+        } userdata = { &taskUploadError, &currentUploadIdx, &filter };
+
+        auto uploadCallback = [](UploadQueueEntry& entry, float progress) {
+            bool has_started = entry.started;
+            bool has_finished = entry.finished;
+            bool has_error = entry.error;
+
+            userdata_upload_t* userdata = (userdata_upload_t*)entry.userdata;
+            bool& taskUploadError = *userdata->taskUploadError;
+            size_t& currentUploadIdx = *userdata->currentUploadIdx;
+            dumpFileFilter& filter = *userdata->filter;
+
+            if(has_started && has_error) {
+                taskUploadError = true;
+                return;
+            }
+
+            if(has_started && has_finished) {
+                currentUploadIdx++;
+                return;
+            }
+
+            WHBLogFreetypeStartScreen();
+            WHBLogFreetypeScreenPrintBottom(L"===============================");
+            WHBLogFreetypeScreenPrintBottom(L"SpotPass Uploader");
+
+            WHBLogFreetypePrintf(L"Uploading SpotPass task file ... (%d out %d)", currentUploadIdx + 1, filter.outMatchedFiles.size());
```

Comment:
I'd recommend using WHBLogPrintf since that should print it to both the screen and the log (it was actually broken but I fixed it today). Then you don't need the print statement below.

Reply:
I'm unsure why this matters, it means every screen draw(around ~50/60 times a second) this message will be logged..